aexpect/remote_door.py

```diff
@@ -0,0 +1,718 @@
+
+    :param str subcontrol: path to the original control
+    :param str list_name: control list name
+    :param value: control list value
+    :type value: [any]
+    :returns: path to the modified control
+    :rtype: str
+
+    .. warning:: The `subcontrol` parameter is control path externally but
+        control content internally after decoration.
+    """
+    return re.sub("%s[ \t\v]*=[ \t\v]*\[.*\]" % list_name.upper(),
+                  "%s = %s" % (list_name.upper(), repr(value)),
+                  subcontrol, count=1)
+
+
+@set_subcontrol
+def set_subcontrol_parameter_dict(subcontrol, dict_name, value):
+    """
+    Replace a dictionary at the beginning of a remote control file.
+
+    :param str subcontrol: path to the original control
+    :param str dict_name: control dictionary name
+    :param value: control dictionary value
+    :type value: {any, any}
+    :returns: path to the modified control
+    :rtype: str
+
+    .. warning:: The `subcontrol` parameter is control path externally but
+        control content internally after decoration.
+    """
+    return re.sub("%s[ \t\v]*=[ \t\v]*\{.*\}" % dict_name.upper(),
+                  "%s = %s" % (dict_name.upper(), repr(value)),
+                  subcontrol, count=1)
+
+
+@set_subcontrol
+def set_subcontrol_parameter_object(subcontrol, value):
+    """
+    Prepare a URI to remote params for the remote control file.
+
+    :param str subcontrol: path to the original control
+    :param value: control parameters value
+    :type value: Params object
+    :returns: path to the modified control
+    :rtype: str
+    :raises: :py:class:`ValueError` if the host IP couldn't be obtained to share the parameters
+
+    .. warning:: The `subcontrol` parameter is control path externally but
+        control content internally after decoration.
+    """
+    params = value
+    host_ip = None
+    nics = params.objects("nics")
+    for nic in nics:
+        nic_params = params.object_params(nic)
+        if nic_params.get("host") is not None:
+            host_ip = nic_params.get("host")
+            break
+    if host_ip is None:
+        raise ValueError("No IP of the host machine was found")
+
+    logging.info("Sharing the test parameters over the network")
+    Pyro4.config.AUTOPROXY = False
+    Pyro4.config.REQUIRE_EXPOSE = False
+    try:
+        pyro_daemon = Pyro4.Daemon(host=host_ip, port=1437)
+        logging.debug("Pyro4 daemon started successfully")
+        uri = pyro_daemon.register(params)
+        pyrod_running = False
+    # address already in use OS error
+    except OSError:
+        pyro_daemon = Pyro4.Proxy("PYRO:" + Pyro4.constants.DAEMON_NAME +
+                                  "@" + host_ip + ":1437")
+        pyro_daemon.ping()
+        registered = pyro_daemon.registered()
+        logging.debug("Pyro4 daemon already started, available objects: %s",
+                      registered)
+        assert(len(registered) == 2)
```

Comment:
I'd add an error message explaining that more than 2 objects are registered, to facilitate debugging.

Reply:
I usually use assertion line included in the stack for this but fair enough. Fixed with next push.